abd-clam/src/core/cluster/tree.rs

```diff
@@ -447,4 +623,88 @@ mod tests {
         // Assert recovering was successful
         assert_trees_equal(&raw_tree, &recovered_tree);
     }
+
+    fn transpose(all_ratios: &Vec<[f64; 6]>) -> Vec<Vec<f64>> {
+        let all_ratios: Vec<f64> = all_ratios.iter().flat_map(|arr| arr.iter().cloned()).collect();
+        let all_ratios: Vec<Vec<_>> = (0..6)
+            .map(|s| all_ratios.iter().skip(s).step_by(6).cloned().collect())
+            .collect();
+
+        all_ratios
+    }
+
+    fn calc_means(all_ratios: &Vec<Vec<f64>>) -> [f64; 6] {
+        let means: [f64; 6] = all_ratios
+            .iter()
+            .map(|values| statistical::mean(values))
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap();
+
+        means
+    }
+
+    fn calc_sds(all_ratios: &Vec<Vec<f64>>) -> [f64; 6] {
+        let means = calc_means(all_ratios);
+        let sds: [f64; 6] = all_ratios
+            .iter()
+            .zip(means.iter())
+            .map(|(values, &mean)| 1e-8 + statistical::population_standard_deviation(values, Some(mean)))
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap();
+        sds
+    }
+
+    #[test]
+    fn test_means() {
+        let all_ratios: Vec<[f64; 6]> = vec![
+            [2.0, 4.0, 5.0, 6.0, 9.0, 15.0],
+            [3.0, 3.0, 6.0, 4.0, 7.0, 10.0],
+            [5.0, 5.0, 8.0, 8.0, 8.0, 1.0],
+        ];
+
+        let transposed = transpose(&all_ratios);
+        let means = calc_means(&transposed);
+
+        let expected_means: [f64; 6] = [3.3333333333333335, 4.0, 6.333333333333334, 6.0, 8.0, 8.666666666666668];
+
+        means.iter().zip(expected_means.iter()).for_each(|(&val1, &val2)| {
+            assert!(
+                (val1 - val2).abs() < 0.001,
+                "Values not equal: val1 = {}, val2 = {}",
+                val1,
+                val2
+            );
+        });
+    }
+
+    #[test]
+    fn test_sds() {
+        let all_ratios: Vec<[f64; 6]> = vec![
+            [2.0, 4.0, 5.0, 6.0, 9.0, 15.0],
+            [3.0, 3.0, 6.0, 4.0, 7.0, 10.0],
+            [5.0, 5.0, 8.0, 8.0, 8.0, 1.0],
+        ];
+
+        let expected_standard_deviations: [f64; 6] = [1.247, 0.816, 1.247, 1.632, 0.816, 5.792];
+        // let sds = calc_sds(&transpose(&all_ratios));
+        let all_ratios: Vec<f64> = all_ratios.iter().flat_map(|arr| arr.iter().cloned()).collect();
+        let all_ratios: Vec<Vec<_>> = (0..6)
+            .map(|s| all_ratios.iter().skip(s).step_by(6).cloned().collect())
+            .collect();
+
+        let sds = calc_sds(&all_ratios);
+
+        sds.iter()
+            .zip(expected_standard_deviations.iter())
+            .for_each(|(&val1, &val2)| {
+                assert!(
+                    (val1 - val2).abs() < 0.001,
+                    "Values not equal: val1 = {}, val2 = {}",
+                    val1,
+                    val2
+                );
```

Comment:
Replace this assert with a relevant macro from the `float_cmp` crate and use a much smaller tolerance.

If the `float_cmp` is not in `Cargo.toml`, add it as a dev dependency.